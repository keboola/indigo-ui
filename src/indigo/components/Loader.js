import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

class Loader extends React.Component {
  render() {
    return <span className={classNames('fa fa-spin fa-spinner', this.props.className)} />;
  }
}

Loader.propTypes = {
  className: PropTypes.string,
};

export default Loader;
