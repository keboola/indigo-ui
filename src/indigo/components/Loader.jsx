import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default React.createClass({
  propTypes: {
    className: PropTypes.string
  },

  render() {
    return (
      <span className={classNames('fa fa-spin fa-spinner', this.props.className)}/>
    );
  }

});
