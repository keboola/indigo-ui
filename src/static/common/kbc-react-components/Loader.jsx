import React from 'react';
import classNames from 'classnames';

export default React.createClass({
  propTypes: {
    className: React.PropTypes.string
  },

  render() {
    return (
      <span className={classNames('fa fa-spin fa-spinner', this.props.className)}/>
    );
  }

});