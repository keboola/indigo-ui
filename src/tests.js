import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
