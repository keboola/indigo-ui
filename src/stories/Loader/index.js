import React from 'react';
import { storiesOf } from '@storybook/react';

import Loader from '../../indigo/components/Loader';

storiesOf('Loader', module).add(
  'Loader',
  () => {
    return <Loader />;
  },
  { info: { text: 'Show if something is loading.' } }
);
