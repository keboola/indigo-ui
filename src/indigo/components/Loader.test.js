import React from 'react';

import { snapshot } from '../../tests';
import Loader from './Loader';

describe('<Loader />', () => {
  it('Loader - default', () => {
    snapshot(<Loader />);
  });

  it('Loader - with classname', () => {
    snapshot(<Loader className="additional-class" />);
  });
});
