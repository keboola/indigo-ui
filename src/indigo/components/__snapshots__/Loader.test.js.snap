// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Loader - default 1`] = `
<i
  className="fa fa-fw fa-spin fa-spinner"
/>
`;

exports[`<Loader /> Loader - with classname 1`] = `
<i
  className="fa fa-fw fa-spin fa-spinner additional-class"
/>
`;
